plugin/src/main/java/org/opensearch/ml/cluster/MLSyncUpCron.java

```diff
@@ -91,6 +129,139 @@ public void run() {
                             ex -> { log.error("Failed to sync model routing", ex); }
                         )
                 );
+
+            // refresh model status
+            if (clusterService.state().getRoutingTable().hasIndex(ML_MODEL_INDEX)) {
+                mlIndicesHandler
+                    .initModelIndexIfAbsent(
+                        ActionListener
+                            .wrap(
+                                res -> { refreshModelState(modelWorkerNodes, loadingModels); },
+                                e -> { log.error("Failed to init model index", e); }
+                            )
+                    );
+            }
         }, e -> { log.error("Failed to sync model routing", e); }));
     }
+
+    @VisibleForTesting
+    void refreshModelState(Map<String, Set<String>> modelWorkerNodes, Map<String, Set<String>> loadingModels) {
```

Comment:
This function is a bit too long. Why not break it into smaller ones?

Reply:
+1. There are lot of `if-else` clauses which are hard to follow. May be add few comments also?

Reply:
Sure, will break to some functions